esphome/components/ld2420/ld2420.cpp

```diff
@@ -0,0 +1,506 @@
+    for (auto &listener : this->listeners_)
+      listener->on_presence(get_presence_());
+  }
+}
+
+void LD2420Component::handle_ack_data_(uint8_t *buffer, int len) {
+  cmd_reply_.command = buffer[CMD_FRAME_COMMAND];
+  cmd_reply_.length = buffer[CMD_FRAME_DATA_LENGTH];
+  uint8_t reg_element = 0;
+  uint8_t data_element = 0;
+  uint16_t data_pos = 0;
+  if (cmd_reply_.length > 64) {
+    ESP_LOGD(TAG, "LD2420 reply - received frame is corrupt, data lenght exceeds 64 bytes.");
+    return;
+  } else if (cmd_reply_.length < 2) {
+    ESP_LOGD(TAG, "LD2420 reply - received frame is corrupt, data lenght is less than 2 bytes.");
+    return;
+  }
+  memcpy(&cmd_reply_.error, &buffer[CMD_ERROR_WORD], sizeof(cmd_reply_.error));
+  const char *result = cmd_reply_.error ? "failure" : "success";
+  if (cmd_reply_.error > 0) {
+    return;
+  };
+  cmd_reply_.ack = true;
+  switch ((uint16_t) cmd_reply_.command) {
+    case (CMD_ENABLE_CONF):
+      ESP_LOGD(TAG, "LD2420 reply - set config enable: CMD = %2X %s", CMD_ENABLE_CONF, result);
+      break;
+    case (CMD_DISABLE_CONF):
+      ESP_LOGD(TAG, "LD2420 reply - set config disable: CMD = %2X %s", CMD_DISABLE_CONF, result);
+      break;
+    case (CMD_READ_REGISTER):
+      ESP_LOGD(TAG, "LD2420 reply - read register: CMD = %2X %s", CMD_READ_REGISTER, result);
+      // TODO Read/Write register is not implemented yet, this will get flushed out to a proper header file
+      data_pos = 0x0A;
+      for (uint16_t index = 0; index < (CMD_REG_DATA_REPLY_SIZE *  // NOLINT
+                                        ((buffer[CMD_FRAME_DATA_LENGTH] - 4) / CMD_REG_DATA_REPLY_SIZE));
+           index += CMD_REG_DATA_REPLY_SIZE) {
+        memcpy(&cmd_reply_.data[reg_element], &buffer[data_pos + index], sizeof(CMD_REG_DATA_REPLY_SIZE));
+        byteswap(cmd_reply_.data[reg_element]);
+        ESP_LOGD(TAG, "Data[%2X]: %s", reg_element, format_hex_pretty(cmd_reply_.data[reg_element]).c_str());
+        reg_element++;
+      }
+      break;
+    case (CMD_WRITE_REGISTER):
+      ESP_LOGD(TAG, "LD2420 reply - write register: CMD = %2X %s", CMD_WRITE_REGISTER, result);
+      break;
+    case (CMD_WRITE_ABD_PARAM):
+      ESP_LOGD(TAG, "LD2420 reply - write gate parameter(s): %2X %s", CMD_WRITE_ABD_PARAM, result);
+      break;
+    case (CMD_READ_ABD_PARAM):
+      ESP_LOGD(TAG, "LD2420 reply - read gate parameter(s): %2X %s", CMD_READ_ABD_PARAM, result);
+      data_pos = CMD_ABD_DATA_REPLY_START;
+      for (uint16_t index = 0; index < (CMD_ABD_DATA_REPLY_SIZE *  // NOLINT
+                                        ((buffer[CMD_FRAME_DATA_LENGTH] - 4) / CMD_ABD_DATA_REPLY_SIZE));
+           index += CMD_ABD_DATA_REPLY_SIZE) {
+        memcpy(&cmd_reply_.data[data_element], &buffer[data_pos + index], sizeof(cmd_reply_.data[data_element]));
+        byteswap(cmd_reply_.data[data_element]);
+        ESP_LOGD(TAG, "Data[%2X]: %s", data_element, format_hex_pretty(cmd_reply_.data[data_element]).c_str());
```

Comment:
Let's maybe make this verbose or even very-verbose along with those below -- otherwise it really makes the logs unnecessarily noisy.

```suggestion
        ESP_LOGVV(TAG, "Data[%2X]: %s", data_element, format_hex_pretty(cmd_reply_.data[data_element]).c_str());
```